pkg/client/request.go

```diff
@@ -0,0 +1,135 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+
+	"github.com/puppetlabs/errawr-go/v2/pkg/encoding"
+	"github.com/puppetlabs/relay/pkg/errors"
+)
+
+func (c *Client) get(path string, headers map[string]string, responseBody interface{}) errors.Error {
```

Comment:
These methods could be a good opportunity to use [option](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) [functions](https://www.sohamkamani.com/golang/options-pattern/). For example, you can't pass a query string the way you have this implemented now. I'd imagine an API kind of like:

```go
Request(
  WithMethod(http.MethodGet),
  WithPath("/foo"),
  WithQuery(url.Values{...}),
  WithResponseInto(&myobj),
)
```

Reply:
Thanks for that, that's a cool pattern